newlib/libc/include/string.h

```diff
@@ -199,7 +199,11 @@ int	 strverscmp (const char *, const char *);
    sure here. */
 #if __GNU_VISIBLE && !defined(basename)
 # define basename basename
+# ifdef __ASMNAME
 char	*__nonnull ((1)) basename (const char *) __asm__(__ASMNAME("__gnu_basename"));
+# else
+char	*__nonnull ((1)) basename (const char *s) { return __gnu_basename(s); }
```

Comment:
This needs to be a #define or a static inline so that it gets compiled correctly.